zkevm-circuits/src/tx_circuit.rs

```diff
@@ -671,19 +724,223 @@ impl<F: Field> SubCircuitConfig<F> for TxCircuitConfig<F> {
         });
 
         ///////////////////////////////////////////////////////////////////////
-        ///////////////  constraints on BlockNum  /////////////////////////////
+        ///////////////  constraints on num_all_txs  // ///////////////////////
+        ///////////////////////////////////////////////////////////////////////
+        meta.create_gate("tx_nonce", |meta| {
+            let mut cb = BaseConstraintBuilder::default();
+
+            cb.condition(is_nonce(meta), |cb| {
+                cb.require_equal(
+                    "tx_nonce = tx_table.value if tag == Nonce",
+                    meta.query_advice(tx_table.value, Rotation::cur()),
+                    meta.query_advice(tx_nonce, Rotation::cur()),
+                );
+            });
+
+            cb.gate(meta.query_fixed(q_enable, Rotation::cur()))
+        });
+
+        meta.create_gate("block_num", |meta| {
```

Comment:
```suggestion
        meta.create_gate("copy block_num", |meta| {
```

Reply:
fixed in https://github.com/scroll-tech/zkevm-circuits/pull/764/commits/bfd23e461fb503ff5fd8a74867829225a6e57fc8